server/protocols/webadmin/webadmin-mailbox-deleted-message-vault/src/test/java/org/apache/james/webadmin/vault/routes/DeletedMessagesVaultRoutesTest.java

```diff
@@ -1849,6 +1917,189 @@ void exportShouldSaveDeletedMessagesDataToBlobStore() throws Exception {
         }
     }
 
+    @Nested
+    class PurgeTest {
+
+        @Test
+        void purgeShouldReturnATaskCreated() {
+            given()
+                .queryParam("action", "restore")
+                .body(MATCH_ALL_QUERY)
+            .when()
+                .post(BOB_PATH)
+            .then()
+                .statusCode(HttpStatus.CREATED_201)
+                .body("taskId", notNullValue());
+        }
+
+        @Test
+        void purgeShouldProduceASuccessfulTaskWithAdditionalInformation() {
+            vault.append(USER, DELETED_MESSAGE, new ByteArrayInputStream(CONTENT)).block();
+            vault.append(USER, DELETED_MESSAGE_2, new ByteArrayInputStream(CONTENT)).block();
+
+            String taskId =
+                given()
+                    .queryParam("action", "purge")
+                .when()
+                    .post()
+                    .jsonPath()
+                    .get("taskId");
+
+            given()
+                .basePath(TasksRoutes.BASE)
+            .when()
+                .get(taskId + "/await")
+            .then()
+                .body("status", is("completed"))
+                .body("taskId", is(taskId))
+                .body("type", is(VaultGarbageCollectionTask.TYPE))
+                .body("additionalInformation.beginningOfRetentionPeriod", is(notNullValue()))
+                .body("additionalInformation.handledUserCount", is(1))
+                .body("additionalInformation.permanantlyDeletedMessages", is(2))
+                .body("additionalInformation.vaultSearchErrorCount", is(0))
+                .body("additionalInformation.deletionErrorCount", is(0))
+                .body("startedDate", is(notNullValue()))
+                .body("submitDate", is(notNullValue()))
+                .body("completedDate", is(notNullValue()));
+        }
+
+        @Test
+        void purgeShouldNotDeleteNotExpiredMessagesInTheVault() throws Exception {
+            MailboxSession session = mailboxManager.createSystemSession(USER.asString());
+            MailboxPath restoreMailboxPath = MailboxPath.forUser(USER.asString(), RESTORE_MAILBOX_NAME);
+            mailboxManager.createMailbox(restoreMailboxPath, session);
+            MessageManager messageManager = mailboxManager.getMailbox(restoreMailboxPath, session);
+            messageManager.appendMessage(
```

Comment:
Why do we need these mailboxManager interactions? I don't see the value it adds in this test. can"t we start it from 

```
            vault.append(USER, DELETED_MESSAGE, new ByteArrayInputStream(CONTENT)).block();
            vault.append(USER, DELETED_MESSAGE_2, new ByteArrayInputStream(CONTENT)).block();
            vault.append(USER, notExpiredMessage, new ByteArrayInputStream(CONTENT)).block();

// ...
```